### ISSUE TITLE
Set-Rendering not saving for other devices than Default

### ISSUE THREAD

**Issue by pierresapinault-ef** [SitecorePowerShell/Console#676]
Hey there,

Using Sitecore 7.2 I am trying to update some datasources that are set from the standard values. We have multiple websites running in the same instance with the same code base so the renderings set on standard values contain tokens that are then replaced with a PS script.

For instance a datasource might be set like this on standard value:
/sitecore/content/sites/$rootname/2017/Home Page/Stage

the scripts looks for all the renderings with a $ and then replaces the tokens accordingly

cd 'master:/sitecore/content/sites/ils'
$Root = Get-Item -Language "en" .
$device = Get-LayoutDevice "Sites2017"

function ReplaceTokens ($item, $device)
{
		$rendering = $item | Get-Rendering -DataSource "*`$*" -Device $device
		$rendering | %{
			$datasource = Get-Item $_.Datasource.Replace([string]"`$rootname",$Root.Name);
			#$_.Datasource = $datasource.ID
			Set-Rendering -Item $item -Instance $_ -DataSource $datasource.ID
		}
}
$Root | % { ReplaceTokens $_ $device }

As you can see i tried 2 different ways, one with setting the datasource ID directly on the rendering definition and the second time by using the Set-Rendering cmdlt. It works perfect when i try on the Default device, but not on any other devices.
Any idea?
best,
Pierre

**Comment by AlexKasaku** [comment]
I've just run some similar script and was able to script it to update the renderings on both the Default and Print devices without any issues. This was using `Set-Rendering`, I don't think it will work with just updating the `Datasource` property as the rendering needs to be removed + reinserted into the item for the update to take place.

This was the script I tested with, on the Home item:

```
$item = Get-Item .
$printDevice = Get-LayoutDevice "Print"

$renderings = Get-Rendering -Item $item -Device $printDevice

$renderings | % {
    Set-Rendering -Item $item -Instance $_ -DataSource $item.ID
}
```

This was however on Sitecore 8.1 - I wouldn't have expected this to make a difference but perhaps it does. 

Could you confirm the version of SPE that you are using?

**Comment by pierresapinault-ef** [comment]
I am using SPE 3.3

**Comment by AlexKasaku** [comment]
I've run the same test in Sitecore 7.2 and it works fine. Looking at the other issue you mentioned, it looks like @AdamNaj made some changes around this area that were released on SPE 4.0, so I would suggest you upgrade to SPE 4.0 to see if this fixes the issue for you.

**Comment by AdamNaj** [comment]
Closing as fixed in 4.0. Verified to be working
